Mathlib/LinearAlgebra/AffineSpace/FiniteDimensional.lean

```diff
@@ -217,6 +230,44 @@ lemma AffineIndependent.card_le_finrank_succ [Fintype ι] {p : ι → P} (hp : A
   exact (affineIndependent_iff_le_finrank_vectorSpan _ _
     (tsub_add_cancel_of_le <| Nat.one_le_iff_ne_zero.2 Fintype.card_ne_zero).symm).1 hp
 
+open Finset in
+/-- If an affine independent finset is contained in the affine span of another finset, then its
+cardinality is at most the cardinality of that finset. -/
+lemma AffineIndependent.card_le_card_of_subset_affineSpan {s t : Finset V}
+    (hs : AffineIndependent k ((↑) : s → V)) (hst : (s : Set V) ⊆ affineSpan k (t : Set V)) :
+    s.card ≤ t.card := by
+  obtain rfl | hs' := s.eq_empty_or_nonempty
+  · simp
+  obtain rfl | ht' := t.eq_empty_or_nonempty
+  · simpa [Set.subset_empty_iff] using hst
+  have := hs'.to_subtype
+  have := ht'.to_set.to_subtype
+  have direction_le := AffineSubspace.direction_le (affineSpan_mono k hst)
+  rw [AffineSubspace.affineSpan_coe, direction_affineSpan, direction_affineSpan,
+    ← @Subtype.range_coe _ (s : Set V), ← @Subtype.range_coe _ (t : Set V)] at direction_le
+  have finrank_le := add_le_add_right (Submodule.finrank_le_finrank_of_le direction_le) 1
+  erw [hs.finrank_vectorSpan_add_one] at finrank_le
```

Comment:
It's a bit evil to have this `erw`. I believe it is due to the syntactic difference between `{x // x ∈ (s : Set V)}` and `{x // x ∈ (s : Finset V)}`. E.g., this works:
```lean
    have foo : finrank k (vectorSpan k (Set.range <| ((↑) : { x // x ∈ (s : Set V) } → V))) + 1 = _ :=
    hs.finrank_vectorSpan_add_one
  rw [foo] at finrank_le
```
Is there an easy way to sidestep this grossness?

Reply:
Yes indeed I used `erw` because of the difference between `{x // x ∈ (s : Set V)}` and `{x // x ∈ (s : Finset V)}`. Why do you think `erw` needs eradicating? It was designed for situations like this one.

Reply:
The thing I don't like about `erw` is that can be abused and it's impossible to tell why it is needed in any given instance so I usually factor it out.

I do however agree that this is reasonable use. If you wish to keep it I suggest adding a comment along the lines of:
```
-- We use `erw` to elide the difference between `{x // x ∈ (s : Set V)}` and `{x // x ∈ (s : Finset V)}`
```
and likewise at the other use below

Reply:
```suggestion
  -- We use `erw` to elide the difference between `↥s` and `↥(s : Set V)}`
  erw [hs.finrank_vectorSpan_add_one] at finrank_le
```